### PR TITLE
Ablom/setup nextjs

### PR DESCRIPTION
**Pull request by andersblom**
# Adds
- NextJS (with a few boilerplate pages)
- Typescript setup for NextJS
- Custom Express server to serve NextJS (with Typescript, we'll see how it goes) so that we can use URL parameters.

### CODE REVIEW

File: server/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import * as next from 'next'
+
+const port: Number = parseInt(<string>process.env.PORT, 10) || 3000
+const dev = process.env.NODE_ENV !== 'production'
+const app = next({ dev })
+const handle = app.getRequestHandler()
+
+app.prepare()
+  .then(() => {
+    const server = express();
```
Comment (andersblom) [inline server/index.ts:11]: Arguably one of the bigger bummers about using SSR if you want URL parameters: You have to serve the app with a custom Express server, to be able to forward URL parameters to React.

On the bright side, it'll be easy to serve without a special sauce web server.

File: server/index.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    server.get('/profile/:id', (req, res) => {
+      /**
+       * How we're going to be showing different pages on the same route conditionally.
+       */
+      if (req.params.id !== 'andersblom' && req.params.id !== 'jerryortega') {
+        return app.render(req, res, '/profile/notFound', { id: req.params.id });
```
Comment (andersblom) [inline server/index.ts:18]: Potentially some cool stuff here. We can validate URL params before they hit the frontend and redirect to a 404 page (or something) before it even hits React.

File: tsconfig.server.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "commonjs",
+    "outDir": "dist",
+    "target": "es2017",
+    "lib": [
+      "es2017"
```
Comment (andersblom) [inline tsconfig.server.json:8]: :partyparrot:

Comment (jeromeportega) [reply tsconfig.server.json:8]: That's pretty cool man.

**Review by jeromeportega — COMMENTED** [review]
There are some missing semi-colons here and there.  Also new lines at the ends of the files.  

On second thought, as part of this PR or as a fast-follow, let's just get prettier up and running.

**Pull request merged**